File: Soutenance 2 End/Miny-RPG Editor/Tiles.vb
Public Class Tiles

    Private Shared This As Tiles

    Public Shared Function [Get](ByVal TileId As Integer) As Bitmap

        If This Is Nothing Then This = New Tiles

        Return This.Tiles(TileId)

    End Function

    Private Tiles As New Dictionary(Of Integer, Bitmap)

    Public Sub New()

        Dim b As New Bitmap("tiles.png")

        Dim i As Integer = 0

        For y As Integer = 0 To b.Height - 1 Step TILE_WIDTH
            For x As Integer = 0 To b.Width - 1 Step TILE_WIDTH

                Dim a As New Bitmap(TILE_WIDTH, TILE_WIDTH)
                For xa As Integer = 0 To TILE_WIDTH - 1
                    For ya As Integer = 0 To TILE_WIDTH - 1
                        Dim c As Color = b.GetPixel(x + xa, y + ya)
                        a.SetPixel(xa, ya, c)
                    Next
                Next
                Tiles.Add(i, a)
                i += 1

            Next
        Next

    End Sub

End Class

File: Soutenance 3/MapEditorReborn/MapEditorReborn/Tiles.cs
using System.Collections.Generic;
using System.Drawing;

namespace MapEditorReborn
{
    public class Tiles
    {
        private static Tiles thisOne;
        private Dictionary<int,Bitmap> tiles = new Dictionary<int, Bitmap>();

        public static Bitmap Get(int tileId)
        {
            if (thisOne == null)
                thisOne = new Tiles();

            return thisOne.tiles[tileId];
        }

        public Tiles()
        {
            Bitmap b = new Bitmap("tiles.png");
            int i = 0;

            for (int y = 0; y < b.Height; y += Constants.TileWidth)
            {
                for (int x = 0; x < b.Width; x += Constants.TileWidth)
                {
                    Bitmap a = new Bitmap(Constants.TileWidth, Constants.TileWidth);

                    for (int xa = 0; xa < Constants.TileWidth; xa++)
                        for (int ya = 0; ya < Constants.TileWidth; ya++)
                        {
                            Color c = b.GetPixel(x + xa, y + ya);
                            a.SetPixel(xa, ya, c);
                        }
                    tiles.Add(i , a);
                    i += 1;
                }
            }

         }
    }
}
